scripts/SANS/sans/gui_logic/presenter/property_manager_service.py

```diff
@@ -0,0 +1,107 @@
+from mantid.api import Algorithm
+from mantid.kernel import (PropertyManagerProperty, PropertyManagerDataService)
+from sans.state.state_base import create_deserialized_sans_state_from_property_manager
+
+
+class PropertyManagerService(object):
+    sans_property_manager_prefix = "SANS_PROPERTY_MANAGER_THIS_NEEDS_TO_BE_UNIQUE_"
+
+    class ConverterAlgorithm(Algorithm):
```

Comment:
The reason for this is convoluted. The `DataProcessorWidget` can only take basic, string-based inputs. The state object which contains all the information required for the reduction cannot be passed into the `DataProcessorWidget`. It was just never intended for more complex reduction setups. We circumvent this limitation by converting the state object into a `PropertyManager` which is then placed into the the `PropertyManagerDataService`. The algorithm which performs the reduction then grabs the state from the `PropertyManagerDataService`. The state object can be serialized into a Python `dict`, but not directly into a `PropertyManager`. The `ConverterAlgorithm` is used to perform this conversion.